src/golang/cmd/server/server/execute_handler.go

```diff
@@ -48,8 +48,11 @@ func HandleSuccess(
 	handlerObj.SendResponse(w, resp)
 }
 
-func ExecuteHandler(server Server, handlerObj handler.Handler) func(w http.ResponseWriter, r *http.Request) {
+func ExecuteHandler(server *AqServer, handlerObj handler.Handler) func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Curious what's the implication of changing this from an interface to a struct. I remember the initial goal for passing in an interface was so that the enterprise edition can reuse this function? If so, maybe we should implement interface APIs in `Server` for updating the atomic bool and mutex and keep the type to be `Server`.

Reply:
We need access to the AqServer's specific fields for locking and releasing the mutex. It feels odd to be exposing a low-level operation such as updating an atomic bool in the `Server` interface. Since we aren't immediately supporting the enterprise edition, can we create a task for now and come back to this? I think we also need to rethink our `Server` interface in general as part of the system internal refactoring push in the next few weeks.

Reply:
I wonder if the enterprise version will have the same fields, so they are not gonna be exclusive to the OSS version, and in that case an interface method seems to make sense. But if we are going to refactor the server interface anyway I don't think this is a blocker for now. Just stamped.

Reply:
True